pkg/api/controllers/gitprovider/branches.go

```diff
@@ -47,7 +49,11 @@ func GetRepoBranches(ctx *gin.Context) {
 
 	response, err := server.GitProviderService.GetRepoBranches(gitProviderId, namespaceId, repositoryId)
 	if err != nil {
-		ctx.AbortWithError(http.StatusInternalServerError, fmt.Errorf("failed to get repo branches: %w", err))
+		statusCode, message, codeErr := controllers.GetHTTPStatusCodeAndMessageFromError(err)
+		if codeErr != nil {
+			ctx.AbortWithError(http.StatusInternalServerError, err)
```

Comment:
We need to abort with `codeErr` instead of `err` (in other controller methods as well)